### PR TITLE
Mt save path 478

### LINKED ISSUE #478: MT Service Prefer Global Training Output
## Is your feature request related to a problem? Please describe.

In the [ModelTrainServicerImpl](https://github.com/caikit/caikit/blob/main/caikit/runtime/servicers/model_train_servicer.py#L107) the `request.training_output_dir` will _always_ point to a local `emptyDir` mount that MT "owns." This works well when the output will be uploaded to S3, but if no S3 output path is given, there's no way to get MT to set this value to a pre-mounted PVC and then not purge the content after the successful training. We need a way to work around this behavior from MT so that user-defined PVCs can be used as the save target for trained models.

## Describe the solution you'd like

In order to support saving to a non-S3 PVC, `caikit` should prefer the globally configured `runtime.training.output_dir` value over `request.training_output_dir` when fielding an MT `Process` request.

## Describe alternatives you've considered

* Save the model to both `runtime.training.output_dir` AND `request.training_output_dir`
* Update MT to better support this usecase

## Additional context

We plan to remove all of this MT functionality from the public `caikit` architecture in short order and do not want to add additional config surface to support this.

### PR DESCRIPTION
**Pull request by gabe-l-hart**
**NOTE**: This is an API breaking change as it will change the behavior of training jobs to result in models failing to be updated unless corresponding changes are made in the server's config to set `runtime.training.output_dir` to `""`.

**What this PR does / why we need it**:

Closes #478 

This PR adjusts the behavior of the Model Train servicer so that if `runtime.training.output_dir` is configured in the global config, the model will be saved there rather than in `request.training_output_dir`. This is needed to support the usecase of saving models to mounted PVCs rather than uploading them to S3 when S3 is not available.

**If applicable**:
- [X] this PR contains unit tests

### CODE REVIEW

**Review by joerunde — APPROVED** [review]
Sweet!

I'm fine with the API breaking change since it only affects users with access to the super-proprietary-and-not-open-sourced `model-train` component. 🤫

File: tests/runtime/servicers/test_model_train_servicer_impl.py [hunk 1]
```diff
@@ -185,38 +185,66 @@ def test_model_train_validation_error_raises(sample_model_train_servicer, output
 
 #####################################################################
 # Normal tests
-def test_model_train_sample_widget(sample_model_train_servicer, output_dir):
+@pytest.mark.parametrize("output_in_config", (True, False))
```
Comment (joerunde) [inline tests/runtime/servicers/test_model_train_servicer_impl.py:188]: very slick tests 👍

File: tests/runtime/servicers/test_model_train_servicer_impl.py [hunk 2]
```diff
@@ -185,38 +185,66 @@ def test_model_train_validation_error_raises(sample_model_train_servicer, output
...
+    output_dir,
+    output_in_config,
+):
+    """This test tests end-to-end training. It includes verifying that the model
+    is saved in the right place. The place where the model is saved comes from:
+
+    1. get_config().runtime.training.output_dir
+    2. request.training_output_dir
+
+    IN THAT ORDER!
```
Comment (joerunde) [inline tests/runtime/servicers/test_model_train_servicer_impl.py:200]: all the more reason that the `process_pb2` handler should move out of caikit/caikit!

Comment (gabe-l-hart) [reply tests/runtime/servicers/test_model_train_servicer_impl.py:200]: 💯

**Pull request merged**